src/lib/operators/sort.cpp

```diff
@@ -63,62 +63,68 @@ class Sort::SortImplMaterializeOutput {
     // copied column by column for each output row. For each column in a row we visit the input column with a reference
     // to the output column. This enables for the SortImplMaterializeOutput class to ignore the column types during the
     // copying of the values.
-    ChunkID chunk_count_out;
-    size_t output_row_count = _row_id_value_vector->size();
-    if (_output_chunk_size) {
-      chunk_count_out = output_row_count % _output_chunk_size ? (output_row_count / _output_chunk_size) + 1
-                                                              : output_row_count / _output_chunk_size;
-    } else {
-      chunk_count_out = 1;
-      _output_chunk_size = output_row_count;
-    }
+    const auto output_row_count = _row_id_value_vector->size();
 
-    auto row_index = 0u;
-
-    for (auto chunk_id_out = ChunkID{0}; chunk_id_out < chunk_count_out; chunk_id_out++) {
-      // Because we want to add the values row wise we have to save all columns temporarily before we can add them to
-      // the
-      // output chunk.
-      auto column_vectors = std::vector<std::shared_ptr<BaseColumn>>(output->column_count());
-      for (ColumnID column_id{0}; column_id < output->column_count(); column_id++) {
-        auto column_type = _table_in->column_type(column_id);
-        column_vectors[column_id] =
-            make_shared_by_data_type<BaseColumn, ValueColumn>(column_type, _table_in->column_is_nullable(column_id));
-      }
+    // Ceiling of integer devision
+    const auto div_ceil = [](auto x, auto y) { return (x + y - 1u) / y; };
+
+    const auto chunk_count_out = div_ceil(output_row_count, _output_chunk_size);
```

Comment:
Naming mismatch: chunk_count_out vs output_row_count. Maybe rename to output_chunk_count?